dlib/dnn/core.h

```diff
@@ -1057,6 +1072,12 @@ namespace dlib
 
         unsigned int sample_expansion_factor() const { return subnet().sample_expansion_factor(); }
 
+        void set_gradient_inputs_to_zero()
+        {
+            x_grad = 0;
```

Comment:
```suggestion
            gradient_input_is_stale = true;
```
This works here too right?

Reply:
Yes, right! Thank you.